portal-impl/src/com/liferay/portal/upgrade/v7_0_0/UpgradePortalPreferences.java

```diff
@@ -83,9 +90,29 @@ protected void upgradeStagingPortalPreferences() throws Exception {
 			while (rs.next()) {
 				long portalPreferencesId = rs.getLong("portalPreferencesId");
 
-				String preferences = rs.getString("preferences");
+				String oldPreferences = rs.getString("preferences");
 
-				ps2.setString(1, convertStagingPreferencesToJSON(preferences));
+				String newPreferences = null;
+
+				try {
+					newPreferences = convertStagingPreferencesToJSON(
+						oldPreferences);
+				}
+				catch (DocumentException de) {
+					_log.error(
+						StringBundler.concat(
+							"PortalPreferences ",
+							String.valueOf(portalPreferencesId),
+							" contains invalid XML, resetting to default"));
+
+					newPreferences = PortletConstants.DEFAULT_PREFERENCES;
+				}
+
+				if (Objects.equals(oldPreferences, newPreferences)) {
+					continue;
+				}
+
+				ps2.setClob(1, new UnsyncStringReader(newPreferences));
```

Comment:
Hey @holatuwol,

In some databases such MySQL we define preferences as a TEXT field. I think we should keep it as setString, is it needed to change to setClob for any reason?

Thanks.